File: sangsoo/0x14 Two pointer/1806.py
tmp = list(map(int, input().split()))
N, S = tmp[0], tmp[1]
arr = list(map(int, input().split()))
prefixsum = [0 for _ in range(N)]

mn = 1e5+1
prefixsum[0] = arr[0]
for i in range(1, N):
    prefixsum[i] = arr[i] + prefixsum[i-1]

st = en = 0
while (st < N):
    while (en < N):
        partialsum = prefixsum[en]-prefixsum[st-1] if st > 0 else prefixsum[en] # C++과 다르게 파이썬에서 arr[-1]하면 뒤에 index를 인식한다. 
        if(partialsum >= S):
            mn = min(mn, en-st)
            break
        en += 1
    st += 1
if(mn == 1e5+1):
    print(0)
else:
    print(mn+1)

File: sangsoo/0x14 Two pointer/1806.cpp
#include <bits/stdc++.h>
using namespace std;

int N, S;
int st, en;
int arr[100002];
int prefixsum[100002];


int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> N >> S;
    for(int i = 0 ; i < N; i++) {
        cin >> arr[i];
    }
    int mn = 1e5+1;
    prefixsum[0] = arr[0];
    for(int i = 1; i < N; i++){
        prefixsum[i] = arr[i] + prefixsum[i-1];
    }
    while(st < N){
        while(en < N){
            if(prefixsum[en]-prefixsum[st-1] >= S){
                mn = min(mn, en-st);
                break;
            }
            en++;
        }
        st++;
    }
    if (mn == 1e5+1) {cout << 0; return 0;}
    cout << mn+1;
}